src/libs/actions/user.ts

```diff
@@ -42,6 +42,54 @@ export async function saveUser(user: User) {
   }
 }
 
+export async function creditUser(clientId: string, credit: number) {
+  try {
+    await db.transaction(async (tx) => {
+      await tx.insert(creditTransactions).values({
+        clientId,
+        credits: credit,
+        paymentId: null,
+      });
+
+      // Update user's remaining balance
+      await tx
+        .update(users)
+        .set({ remainingBalance: 150 })
+        .where(eq(users.clientId, clientId));
+    });
+    return { success: true };
+  } catch (error) {
+    console.error('Error crediting new user:', error);
+    return { error: 'Failed to credit user.' };
+  }
+}
+
+export async function deductCredit(clientId: string) {
+  try {
+    // Fetch user's current balance
+    const user = await db
+      .select({ remainingBalance: users.remainingBalance })
+      .from(users)
+      .where(eq(users.clientId, clientId))
+      .limit(1);
+
+    if (!user.length || (user[0]?.remainingBalance as number) <= 0) {
+      return { error: 'Insufficient credits.' };
```

Comment:
errors should always be thrown and not returned.
Or else, the caller will always have to have error or success checks, which is not right